p2p/net/swarm/swarm_transport.go

```diff
@@ -19,25 +19,23 @@ func (s *Swarm) TransportForDialing(a ma.Multiaddr) transport.Transport {
 
 	s.transports.RLock()
 	defer s.transports.RUnlock()
+
 	if len(s.transports.m) == 0 {
 		// make sure we're not just shutting down.
 		if s.transports.m != nil {
 			log.Error("you have no transports configured")
 		}
 		return nil
 	}
-
-	for _, p := range protocols {
-		transport, ok := s.transports.m[p.Code]
-		if !ok {
-			continue
-		}
-		if transport.Proxy() {
-			return transport
+	if isRelayAddr(a) {
```

Comment:
I don't really like this logic here. Instead, we could have transports reject `/p2p-circuit` address. wdyt?

Reply:
Let's not change this just yet since that would involve changing every transport. Let's revisit after discussing https://github.com/multiformats/multiaddr/pull/140